mlflow/utils/rest_utils.py

```diff
@@ -267,6 +269,8 @@ def get_workspace_client(
         kwargs = {"host": host, "token": token}
     else:
         kwargs = {"profile": databricks_auth_profile}
+    if timeout:
```

Comment:
```suggestion
    if timeout is not None:
```

because `0` is falsy.